python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -207,6 +208,14 @@ def trigger_out_of_band_tensor_transfer(
                 # If the source and destination actors are the same, the tensors can
                 # be transferred intra-process, so we skip the out-of-band tensor
                 # transfer.
+                if src_actor == dst_actor:
```

Comment:
Can you update this so that we only warn if we go through this codepath and we also pass the same ref to another actor? Passing to self is okay, the issue is if we pass to self and to another actor.

We should also update to only warn once per application.